src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -179,83 +221,121 @@ function evaluateBinaryLiteral(node: BinaryOperation): RationalLiteral | boolean
-  } else if (typeof left !== 'boolean' && typeof right !== 'boolean') {
+    ast.replaceNode(node, result.node);
+    return result;
+  } else if (typeof left.evaluation !== 'boolean' && typeof right.evaluation !== 'boolean') {
+    let result;
     switch (node.operator) {
       case '**':
-        return left.exp(right);
+        result = generateInfo(left.evaluation.exp(right.evaluation), node, ast);
+        break;
       case '*':
-        return left.multiply(right);
+        result = generateInfo(left.evaluation.multiply(right.evaluation), node, ast);
+        break;
       case '/':
-        return left.divideBy(right);
+        result = generateInfo(left.evaluation.divideBy(right.evaluation), node, ast);
+        break;
       case '%':
-        return left.mod(right);
+        result = generateInfo(left.evaluation.mod(right.evaluation), node, ast);
+        break;
       case '+':
-        return left.add(right);
+        result = generateInfo(left.evaluation.add(right.evaluation), node, ast);
+        break;
       case '-':
-        return left.subtract(right);
+        result = generateInfo(left.evaluation.subtract(right.evaluation), node, ast);
+        break;
       case '>':
-        return left.greaterThan(right);
+        result = generateInfo(left.evaluation.greaterThan(right.evaluation), node, ast);
+        break;
       case '<':
-        return right.greaterThan(left);
+        result = generateInfo(right.evaluation.greaterThan(left.evaluation), node, ast);
+        break;
       case '>=':
-        return !right.greaterThan(left);
+        result = generateInfo(!right.evaluation.greaterThan(left.evaluation), node, ast);
+        break;
       case '<=':
-        return !left.greaterThan(right);
+        result = generateInfo(!left.evaluation.greaterThan(right.evaluation), node, ast);
+        break;
       case '==':
-        return left.equalValueOf(right);
+        result = generateInfo(left.evaluation.equalValueOf(right.evaluation), node, ast);
+        break;
       case '!=':
-        return !left.equalValueOf(right);
+        result = generateInfo(!left.evaluation.equalValueOf(right.evaluation), node, ast);
+        break;
       case '<<':
-        return left.shiftLeft(right);
+        result = generateInfo(left.evaluation.shiftLeft(right.evaluation), node, ast);
+        break;
       case '>>':
-        return left.shiftRight(right);
+        result = generateInfo(left.evaluation.shiftRight(right.evaluation), node, ast);
+        break;
       case '&':
-        return left.bitwiseAnd(right);
+        result = generateInfo(left.evaluation.bitwiseAnd(right.evaluation), node, ast);
+        break;
       case '|':
-        return left.bitwiseOr(right);
+        result = generateInfo(left.evaluation.bitwiseOr(right.evaluation), node, ast);
+        break;
       case '^':
-        return left.bitwiseXor(right);
+        result = generateInfo(left.evaluation.bitwiseXor(right.evaluation), node, ast);
+        break;
```

Comment:
Imo it would be more readable:
```
opToFunction = {
 "|": left.evaluation.bitwiseOr,
  ...
}
result = generateInfo(opToFunction[node.operator](right.evaluation), node, ast)
```

Reply:
It would probbably require `"|": left.evaluation.bitwiseOr.bind(left.evaluation)` or something though

Reply:
We've got higher priorities right now, let's merge is as it is